### PR TITLE
more details to the documentation of data preprocessing

### PR DESCRIPTION
**Pull request by Mivg**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -98,11 +98,11 @@ To get started with DCLM, follow these steps:
     We recommend the use of Python 3.10 with DCLM.
 
 ## Selecting Raw Sources
-If you are creating a new source:
+If you are creating a new source (for example, Wikipedia, GitHub, etc.):
```
Comment (GeorgiosSmyrnis) [inline README.md:101]: Maybe creating -> registering?

File: README.md [hunk 2]
```diff
@@ -98,11 +98,11 @@ To get started with DCLM, follow these steps:
...
 
-- Ensure your data is stored in JSONL format (ideally compressed with zstandard).
-- Key names should be consistent with those in [here](baselines/core/constants.py).
-- Create a reference JSON in [exp_data/datasets/raw_sources](exp_data/datasets/raw_sources).
+- Ensure your data is stored in JSONL format (ideally compressed with zstandard), where each line correspond to single page.
+- Key names in these JSONL should be consistent with those in [here](baselines/core/constants.py).
```
Comment (GeorgiosSmyrnis) [inline README.md:104]: Maybe we can explain these in bit more detail here?

File: README.md [hunk 3]
```diff
@@ -119,6 +119,7 @@ To process raw data, follow these steps:
 
 2. **Set up a Ray cluster**:
     The data processing script relies on Ray for distributed processing of data. This cluster can be either launched on a single node (for small scale data processing) or using AWS EC2 instances.
+    There is also work to [deploy Ray on slurm setups](https://docs.ray.io/en/latest/cluster/vms/user-guides/community/slurm.html), though this effort is still a work-in-progres.
```
Comment (GeorgiosSmyrnis) [inline README.md:122]: I feel like this isn't really WIP anymore, in the sense that it is possible to do so if you follow the guide.

Comment (jeffreywpli) [reply README.md:122]: has one of us been able to get our scripts to work on slurm?

Comment (GeorgiosSmyrnis) [reply README.md:122]: I've been launching Ray jobs on TACC, so that part is stable - but I suppose you're right, best not to claim something that might be false

File: README.md [hunk 4]
```diff
@@ -110,28 +110,30 @@ The jsons get used in the following way:
...
+
 ## Selecting Raw Sources
-If you are creating a new source:
+If you are creating and registering a new source (for example, Wikipedia, GitHub, etc.):
 
-- Ensure your data is stored in JSONL format (ideally compressed with zstandard).
-- Key names should be consistent with those in [here](baselines/core/constants.py).
-- Create a reference JSON in [exp_data/datasets/raw_sources](exp_data/datasets/raw_sources).
+- Ensure your data is stored in JSONL format, ideally compressed with zstandard (though uncompressed or gzip-compressed files will also work), where each line corresponds to a single page/document.
+- Key names in these JSONL should be consistent with those in [here](baselines/core/constants.py). Most importantly, there should be a ``"text"`` key for each line that contains the acftual content of the page.
```
Comment (GeorgiosSmyrnis) [inline README.md:118]: typo: acftual - > actual

Comment (GeorgiosSmyrnis) [inline README.md:118]: I would rephrase as follows:
> Each row in these JSONL files corresponds to a document. Each row should contain keys consistent with those at ... and at minimum should contain at least a "text" key containing the actual content.
